modules/tensor_mechanics/src/actions/CohesiveZoneMasterAction.C

```diff
@@ -23,41 +25,100 @@ CohesiveZoneMasterAction::validParams()
   params.addRequiredParam<std::vector<BoundaryName>>(
       "boundary", "The list of boundary IDs from the mesh where the cohesive zone will be applied");
   params.addRequiredParam<std::vector<VariableName>>(
-      "displacements",
-      "The displacements appropriate for the simulation geometry and coordinate system");
+      "displacements", "The nonlinear displacement variables for the problem");
+  MooseEnum kinematicType("SmallStrain TotalLagrangian", "SmallStrain");
+  params.addParam<MooseEnum>("kinematic", kinematicType, "Kinematic formulation");
+  params.addParam<std::string>("base_name", "Material property base name");
   return params;
 }
 
-CohesiveZoneMasterAction::CohesiveZoneMasterAction(const InputParameters & params) : Action(params)
+CohesiveZoneMasterAction::CohesiveZoneMasterAction(const InputParameters & params)
+  : Action(params),
+    _displacements(getParam<std::vector<VariableName>>("displacements")),
+    _ndisp(_displacements.size()),
+    _base_name(isParamValid("base_name") && !getParam<std::string>("base_name").empty()
+                   ? getParam<std::string>("base_name")
+                   : ""),
+    _kinematic(getParam<MooseEnum>("kinematic").getEnum<Kinematic>())
 {
+  // We can't enforce conssintency between the number of displacement varaiables and the mesh
+  // dimension. Hence we only check we have a reasonable number of displacement variables
+  if (_ndisp > 3 || _ndisp < 1)
+    mooseError("the CZM Action requires 1, 2 or 3 displacement variables.");
```

Comment:
I am curious whether TM action checks this.

Reply:
If I remember correctly it does not. Is there any reason why you don't enforce this?